source/common/router/router.cc

```diff
@@ -1352,10 +1406,16 @@ void Filter::onUpstreamHeaders(uint64_t response_code, Http::ResponseHeaderMapPt
       // We already retried this request (presumably for a per try timeout) so
       // we definitely won't retry it again. Check if we would have retried it
       // if we could.
-      could_not_retry = retry_state_->wouldRetryFromHeaders(*headers);
+      bool dumb;
```

Comment:
Instead of naming this "dumb", how about just naming it retry_as_early_data and then perhaps commenting about how it's not being used?